numpy/core/tests/test_item_selection.py

```diff
@@ -68,6 +68,21 @@ def test_unicode_mode(self):
         k = b'\xc3\xa4'.decode("UTF8")
         assert_raises(ValueError, d.take, 5, mode=k)
 
+    def test_partition(self):
+        # In reference to github issue #6530
+        a = np.array([0, 2, 4, 6, 8, 10])
+        a = a.partition(np.array([], dtype=np.int16))
+
+        assert_equal(a, None)
```

Comment:
sorry, that should be `assert_array_equal` (and then you can remove the newly-added import of `assert_equal` at the top of the file)